laws/src/main/scala/cats/laws/TraverseLaws.scala

```diff
@@ -129,6 +129,43 @@ trait TraverseLaws[F[_]] extends FunctorLaws[F] with FoldableLaws[F] with Unorde
     val rhs = F.map(F.mapWithIndex(fa)((a, i) => (a, i)))(f)
     lhs <-> rhs
   }
+
+  def mapWithLongIndexRef[A, B](fa: F[A], f: (A, Long) => B): IsEq[F[B]] = {
```

Comment:
@nikololiahim I just realized, we added the laws but not the tests 😅 I completely forgot about those, so sorry!

https://github.com/typelevel/cats/blob/094280363220d7d0e57c7be4e00c3847ba2188e1/laws/src/main/scala/cats/laws/discipline/TraverseTests.scala#L74-L83

Reply:
Should I reopen and add them here? Or open a separate PR?

Reply:
Whatever works, although I'm not sure if it's possible to re-open a merged PR :) thanks!!